DEVELOPERS.rst

```diff
@@ -0,0 +1,68 @@
+For developers of ZODB
```

Comment:
Well, it's published here.

+1 to adding a repo URL to the readme and making it more explicit in the docs.

Eventually, I think development should be a topic in the docs.

This PR just aims to unscrew some things.

At a meta level, If my goal was to be perfect, I would probably have put this off and the PyPI page would have remained broken longer.